src/shopgate/gambiogx/includes/modules/payment/ShopgateInstallHelper.php

```diff
@@ -119,19 +119,27 @@ private function getUid()
             }
         }
 
-        $keyQuery =
-            'SELECT c.configuration_value AS val FROM ' . TABLE_CONFIGURATION . ' AS c WHERE c.configuration_key = "'
-            . self::SHOPGATE_CALLBACK_DATABASE_CONFIG_KEY . '" LIMIT 1;';
+        if ($this->useLegacyConfigTable()) {
+            $keyQuery = 'SELECT c.configuration_value AS val FROM ' . TABLE_CONFIGURATION . ' AS c
+                         WHERE c.configuration_key = "' . self::SHOPGATE_CALLBACK_DATABASE_CONFIG_KEY . '" LIMIT 1;';
+        } else {
+            $keyQuery = 'SELECT c.`value` AS val FROM ' . TABLE_CONFIGURATION . ' AS c
+                         WHERE c.`key` = "' . self::SHOPGATE_CALLBACK_DATABASE_CONFIG_KEY . '" LIMIT 1;';
+        }
         $result   = xtc_db_query($keyQuery);
         $row      = xtc_db_fetch_array($result);
 
         if (!empty($row) && $row['val'] && $row['val'] != 0) {
             return $row['val'];
         }
 
-        $tokenQuery = 'SELECT c.gm_value as token FROM gm_configuration as c WHERE c.gm_key = "SECURE_TOKEN" LIMIT 1;';
-        $result     = xtc_db_query($tokenQuery);
-        $row        = xtc_db_fetch_array($result);
+        try {
+            $tokenQuery = 'SELECT c.gm_value as token FROM gm_configuration as c WHERE c.gm_key = "SECURE_TOKEN" LIMIT 1;';
+            $result     = xtc_db_query($tokenQuery);
+            $row        = xtc_db_fetch_array($result);
+        } catch (Exception $e) {
+            $row = [];
```

Comment:
As per PHP CS Fixer:

```suggestion
            $row = array();
```